File: backend/app/schemas/airport.py
"""
Airport schemas for request/response validation.
"""
from pydantic import BaseModel, Field


class AirportResponse(BaseModel):
    """Airport response."""
    id: int
    code: str = Field(max_length=3)  # IATA code
    name: str
    city: str
    country: str
    timezone: str | None
    
    class Config:
        from_attributes = True

File: AIT SOLUTION/mobile_app/lib/features/flight/domain/entities/airport.dart
import 'package:equatable/equatable.dart';

class Airport extends Equatable {
  final int id;
  final String iataCode;
  final String name;
  final String city;
  final String country;
  final String timezone;

  const Airport({
    required this.id,
    required this.iataCode,
    required this.name,
    required this.city,
    required this.country,
    required this.timezone,
  });

  @override
  List<Object?> get props => [id, iataCode, name, city, country, timezone];
}
